hypertrace-trace-enricher/trace-reader/src/main/java/org/hypertrace/trace/accessor/entities/DefaultTraceEntityAccessor.java

```diff
@@ -171,22 +157,31 @@ private boolean passesFormationCondition(Entity entity, EntityFormationCondition
     }
   }
 
-  private Maybe<Map<String, AttributeValue>> resolveAllAttributes(
+  private Optional<Map<String, AttributeValue>> resolveAllAttributes(
       String scope, StructuredTrace trace, Event span) {
-    return spanTenantContext(span)
-        .wrapSingle(() -> this.attributeClient.getAllInScope(scope))
-        .flattenAsObservable(list -> list)
-        .filter(this::isEntitySourced)
-        .flatMapMaybe(attributeMetadata -> this.resolveAttribute(attributeMetadata, trace, span))
-        .collect(Collectors.toMap(Entry::getKey, Entry::getValue))
-        .toMaybe();
+    List<AttributeMetadata> attributeMetadataList =
+        this.attributeClient.getAllInScope(
+            this.traceAttributeReader.getRequestContext(span), scope);
+    if (attributeMetadataList.isEmpty()) {
+      return Optional.empty();
```

Comment:
this short circuit isn't really necessary and doesn't significantly lend readability.